scheduler/scheduler.go

```diff
@@ -187,6 +215,9 @@ func BuildDependencyGraph(c client.Interface, sel labels.Selector) (DependencyGr
 			depGraph[d.Child].Requires, depGraph[parent])
 		depGraph[parent].RequiredBy = append(
 			depGraph[parent].RequiredBy, depGraph[child])
+		if err := depGraph[parent].updateMeta(d.Meta); err != nil {
```

Comment:
This code looks like meta is only related to parent. I'm ok with the approach, but that might not be exactly understood by user. We should make a bold statement about it and document it.